### PR TITLE
Drop Python 2.7.

### LINKED ISSUE #71: dropping python 2.7
I thought I write down my thoughts on this backport w.r.t. dropping Python 2.7 as this might need some discussion.

The current code for pathlib2 contains a lot of extremely ugly (and type-annotation unfriendly) hacks relating to Python 2.7. If we drop it, there's no need to keep those hacks around, resulting in cleaner code with less potential sources for bugs, and with more helpful type annotations. So, instead of manually removing those hacks, I'd actually suggest to make a fresh backport by taking the current cpython pathlib module and simply patching it to work on 3.6+. I believe this will be the easiest route going forward.

### PR DESCRIPTION
**Pull request by mcmtroffaes**
Copied fresh pathlib.py and test_pathlib.py files from cpython, and updated to work on all versions of Python starting from 3.6.

Fixes #71 and in a way supersedes #73.

### CODE REVIEW

**Comment by mcmtroffaes** [comment]
A heads-up that I'll be merging this shortly, since 2.3.7 (the last version to still support Python 2.7) has just been released.

File: tests/compat_contextlib.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from contextlib import AbstractContextManager
+
+
+class nullcontext(AbstractContextManager):
```
Comment (graingert) [inline tests/compat_contextlib.py:4]: Might be worth using contextlib2.nullcontext?

Comment (mcmtroffaes) [reply tests/compat_contextlib.py:4]: Ah yes, could do. It's just for the tests so I did a very quick copy-paste from cpython, but can use contextlib2 instead. Will follow up tomorrow.

Comment (mcmtroffaes) [reply tests/compat_contextlib.py:4]: Updated here: https://github.com/jazzband/pathlib2/commit/2c8f4b1b842efce9fda4ca67c2b1b0633bedc052

**Pull request merged**